linalg/operator.hpp

```diff
@@ -433,6 +433,81 @@ class ConstrainedOperator : public Operator
    virtual ~ConstrainedOperator() { if (own_A) { delete A; } }
 };
 
+
+/** @brief Mimic the action of a complex operator using two real operators.
+
+    This operator requires vectors that are twice the length of its internally
+    stored real operators, Op_Real and Op_Imag. It is assumed that these vectors
+    store the real part of the vector first followed by its imaginary part.
+
+    ComplexOperator allows one to choose a convention upon construction, which
+    facilitates symmetry.
+
+    Matrix-vector products are then computed as:
+
+    1. When Convention::HERMITIAN is used (default)
+    / y_r \   / Op_r -Op_i \ / x_r \
+    |     | = |            | |     |
+    \ y_i /   \ Op_i  Op_r / \ x_i /
+
+    2. When Convention::BLOCK_SYMMETRIC is used
+    / y_r \   / Op_r -Op_i \ / x_r \
+    |     | = |            | |     |
+    \-y_i /   \-Op_i -Op_r / \ x_i /
+
+    Either convention can be used with a given complex operator,
+    however, each of them is best suited for certain classes of
+    problems.  For example:
+
+    1. Convention::HERMITIAN, is well suited for Hermitian operators,
+    i.e. operators where the real part is symmetric and the imaginary part of
+    the operator is anti-symmetric, hence the name. In such cases the resulting
+    2 x 2 operator will be symmetric.
+
+    2. Convention::BLOCK_SYMMETRIC, is well suited for operators where both the
+    real and imaginary parts are symmetric. In this case the resulting 2 x 2
+    operator will again be symmetric. Such operators are common when studying
+    damped oscillations, for example.
+ */
```

Comment:
It is probably worth pointing out that general _nonlinear_ complex operators cannot be represented by this class, because it assumes that `Re(A(x)) = A_r(x_r) - A_i(x_i)` and `Im(A(x)) = A_r(x_i) + A_i(x_r)` for some real (nonlinear) operators `A_r` and `A_i`.